frontend/src/employee-mobile-frontend/child-attendance/actions/MarkReservations.tsx

```diff
@@ -0,0 +1,540 @@
+  reservation: NonReservableReservation
+}) => {
+  const { i18n } = useTranslation()
+
+  return (
+    <div data-qa="reservation-text">
+      {reservation.absenceType !== null ? (
+        <ServiceTime>{i18n.attendances.types.ABSENT}</ServiceTime>
+      ) : (
+        <AttendanceDailyServiceTimes
+          hideLabel={true}
+          times={reservation.dailyServiceTimes}
+          reservations={reservation.reservations}
+        />
+      )}
+    </div>
+  )
+}
+
+const ReservationsEdit = ({
+  childId,
+  reservations: data,
+  onCancel,
+  onSuccess
+}: {
+  childId: UUID
+  reservations: NonReservableReservation[]
+  onCancel: () => void
+  onSuccess: () => void
+}) => {
+  const { i18n, lang } = useTranslation()
+  const form = useForm(
+    reservationsForm,
+    () => initialFormState(data),
+    i18n.attendances.validationErrors
+  )
+  const { reservations } = useFormFields(form)
+  const boundReservations = useFormElems(reservations)
+
+  const minDatesByWeek = useMemo(
+    () => getMinDatesByWeek(reservations.state),
+    [reservations]
+  )
+
+  return (
+    <>
+      {boundReservations.map((reservation) => (
+        <React.Fragment key={reservation.state.date.format()}>
+          {minDatesByWeek[reservation.state.date.getIsoWeek()] ===
+            reservation.state.date && (
+            <Label primary>
+              {i18n.common.week} {reservation.state.date.getIsoWeek()}
+            </Label>
+          )}
+          <FixedSpaceRow
+            alignItems="baseline"
+            data-qa={`reservation-date-${reservation.state.date.formatIso()}`}
+          >
+            <Label>{reservation.state.date.format('EEEEEE d.M.', lang)}</Label>
+            <ReservationEdit form={reservation} />
+          </FixedSpaceRow>
+          <Gap size="s" />
+        </React.Fragment>
+      ))}
+      <HorizontalLine slim />
+      <Actions>
+        <FixedSpaceRow fullWidth>
+          <Button text={i18n.common.cancel} onClick={onCancel} />
+          <MutateButton
+            primary
+            text={i18n.common.confirm}
+            mutation={setNonReservableReservationsMutation}
+            onClick={() => ({
+              childId,
+              body: reservations.value()
+            })}
+            onSuccess={onSuccess}
+            disabled={!form.isValid()}
+            data-qa="confirm"
+          />
+        </FixedSpaceRow>
+      </Actions>
+    </>
+  )
+}
+
+const ReservationEdit = ({
+  form
+}: {
+  form: BoundForm<typeof reservationForm>
+}) => {
+  const { times, absenceType } = useFormFields(form)
+  const boundTimes = useFormElems(times)
+
+  const formInputInfo = form.inputInfo()
+  const addButtonLocation =
+    form.isValid() &&
+    form.state.times.every(
+      (reservation) =>
+        reservation.startTime !== '' && reservation.endTime !== ''
+    )
+      ? boundTimes.length === 1
+        ? 'same-row'
+        : boundTimes.length > 1
+        ? 'next-row'
+        : undefined
+      : undefined
```

Comment:
The maximum amount of reservations per day is 2. This is not enforced by constraints or such, but I think it is assumed somewhere in the backend logic. A citizen cannot add more than 2 reservations anyway.